ProjFS.Mac/PrjFSKextTests/MockVnodeAndMount.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include "../PrjFSKext/kernel-header-wrappers/vnode.h"
+#include "../PrjFSKext/kernel-header-wrappers/mount.h"
+#include "../PrjFSKext/public/FsidInode.h"
+#include <memory>
+#include <string>
+
+typedef std::shared_ptr<mount> MountPointer;
+typedef std::shared_ptr<vnode> VnodePointer;
+typedef std::weak_ptr<vnode> VnodeWeakPointer;
+
+struct mount
+{
+private:
+    vfsstatfs statfs;
+    uint64_t nextInode;
+    
+public:
+    static MountPointer Create(const char* fileSystemTypeName, fsid_t fsid, uint64_t initialInode);
+    
+    inline fsid_t GetFsid() const { return this->statfs.f_fsid; }
+    
+    friend struct vnode;
+    friend vfsstatfs* vfs_statfs(mount_t mountPoint);
+};
+
+struct vnode
+{
+private:
+    VnodeWeakPointer weakSelfPointer;
+    MountPointer mountPoint;
+    
+public:
+    bool isRecycling = false;
+    vtype type = VREG;
+
+private:
+    uint64_t inode;
```

Comment:
For my own understanding can you explain how we layout the types here.  I see we go private/public/private/public.  I want to ensure I understand the standard we have.

Reply:
I've been grouping the member variables logically rather than by access permissions. The 2 remaining public ones are probably better off private and accessed via functions, though.